### PR TITLE
#2976. Add more grammar, semantics and type inference test

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Static-access-shorthand/semantics_A02_t01.dart [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+/// `.id(args)`, `.id<typeArgs>(args)`, `.new` or `.new(args)`.
+/// ...
+/// The general rule is that any of the expression forms above, starting with
+/// `.id`, are treated exactly as if they were preceded by a fresh prefixed
+/// identifier `_p.C` which denotes the declaration of the type of the context
+/// type scheme of the entire `<staticMemberShorthand>`.
+///
+/// @description Checks that it is a compile-time error if in a shorthand
+/// expression of the form `.id`, `.id()` or `.id<...>()` the type of the
+/// context doesn't declare member named `id`.
```
Comment (eernstg) [inline LanguageFeatures/Static-access-shorthand/semantics_A02_t01.dart]: Typo:
```suggestion
/// context doesn't declare a member named `id`.
```

**Review by eernstg — APPROVED** [review]
LGTM

File: LanguageFeatures/Static-access-shorthand/semantics_A02_t01.dart (hunk 1 above)
Comment (eernstg) [reply LanguageFeatures/Static-access-shorthand/semantics_A02_t01.dart]: (I fixed this, because there were no other comments)

Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/semantics_A02_t01.dart]: Thank you!

**Pull request merged**